### ISSUE TITLE
Dependent nuget packages are not listed

### ISSUE THREAD

**Issue by quasarea** [madushans/DependenSee#4]
tried with sample .net 5.0 project, added newtonsoft.json as dependency, built but 
```
PS C:\git\.Azure> DependenSee.exe -s .\ConsoleApp\ -t consolejson -IncludePackages
```
does 
```

{
  "Projects": [
    {
      "Id": "ConsoleApp\\ConsoleApp.csproj",
      "Name": "ConsoleApp"
    },
    {
      "Id": "ConsoleApp.Tests\\ConsoleApp.Tests.csproj",
      "Name": "ConsoleApp.Tests"
    }
  ],
  "Packages": [],
  "References": []
}
```

**Comment by madushans** [comment]
Hi @quasarea

I can repro this as you mentioned.

It seems like this is result of the way Visual Studio manages the project. The project changes are not written to the csproj file immediately.

You have to either build the solution or close VS for it to flush the csproj file contents to disk. DependenSee cannot see the changes until this happens.

To confirm, you can 
- open the csproj file in notepad++ (do not use notepad as it can lock the file) 
- open solution in VS
- Add a package from nuget
- switch back to notepad++
- you should not get the dialog saying the file has changed
- Build the solution or close VS
- switch back to notepad++
- you should now get the dialog asking to reload the file from disk

I believe if you wait long enough with no input to VS, it will eventually flush the csproj files when the system is idle, so you have to do the above relatively quickly to reproduce the issue.

Please let me know if this helps. If you're still experiencing the issue, also include the csproj file in question. I can have a deeper look.

Thanks for using DependenSee.

**Comment by quasarea** [comment]
Hi, thanks for quick reply, unfortunately it looks like problem is not related to vs slowness. I was doing it from console, with no vs open at the time. Also assumed you would be using Solution\Project\obj\Project\project.assets.json file to detect dependencies.

Example case:
![image](https://user-images.githubusercontent.com/10334982/120986014-10fbc000-c774-11eb-92d5-795d02eb99b1.png)

Output (same result on bare solution, nuget restored and fully built):
```
DependenSee.exe -s .\VsTestIssue\ -t ConsoleJson
{
  "Projects": [
    {
      "Id": "DotNetStandardLibrary\\DotNetStandardLibrary.csproj",
      "Name": "DotNetStandardLibrary"
    },
    {
      "Id": "DotNetStandardLibrary.Tests\\DotNetStandardLibrary.Tests.csproj",
      "Name": "DotNetStandardLibrary.Tests"
    },
    {
      "Id": "DotNetStandardLibrary.XUnit.Tests\\DotNetStandardLibrary.XUnit.Tests.csproj",
      "Name": "DotNetStandardLibrary.XUnit.Tests"
    }
  ],
  "Packages": [],
  "References": [
    {
      "From": "DotNetStandardLibrary.Tests\\DotNetStandardLibrary.Tests.csproj",
      "To": "DotNetStandardLibrary\\DotNetStandardLibrary.csproj"
    },
    {
      "From": "DotNetStandardLibrary.XUnit.Tests\\DotNetStandardLibrary.XUnit.Tests.csproj",
      "To": "DotNetStandardLibrary\\DotNetStandardLibrary.csproj"
    }
  ]
}
```

Solution to replicate [VsTestIssue.zip](https://github.com/madushans/DependenSee/files/6607286/VsTestIssue.zip)

I also attached example assets file, you actually need to build or at least nuget restore to get this file generated. [project.assets.json.zip](https://github.com/madushans/DependenSee/files/6607307/project.assets.json.zip)

**Comment by madushans** [comment]
In your command, you're missing the `-IncludePackages` flag.


Instead of `DependenSee.exe -s .\VsTestIssue\ -t ConsoleJson`

use `DependenSee.exe -s .\VsTestIssue\ -t ConsoleJson -P`

or `DependenSee.exe -s .\VsTestIssue\ -t ConsoleJson -IncludePackages`

DependenSee excludes packages unless specified otherwise, to prevent HTML output from being overwhelming to look at.

I have downloaded the VsTestIssue.zip and ran DependenSee without building the solution or opening it in VS. 

Below is my output.

```json
{
  "Projects": [
    {
      "Id": "VsTestIssue\\DotNetStandardLibrary\\DotNetStandardLibrary.csproj",
      "Name": "DotNetStandardLibrary"
    },
    {
      "Id": "VsTestIssue\\DotNetStandardLibrary.Tests\\DotNetStandardLibrary.Tests.csproj",
      "Name": "DotNetStandardLibrary.Tests"
    },
    {
      "Id": "VsTestIssue\\DotNetStandardLibrary.XUnit.Tests\\DotNetStandardLibrary.XUnit.Tests.csproj",
      "Name": "DotNetStandardLibrary.XUnit.Tests"
    }
  ],
  "Packages": [
    {
      "Id": "Newtonsoft.Json",
      "Name": "Newtonsoft.Json"
    },
    {
      "Id": "Microsoft.NET.Test.Sdk",
      "Name": "Microsoft.NET.Test.Sdk"
    },
    {
      "Id": "MSTest.TestAdapter",
      "Name": "MSTest.TestAdapter"
    },
    {
      "Id": "MSTest.TestFramework",
      "Name": "MSTest.TestFramework"
    },
    {
      "Id": "coverlet.collector",
      "Name": "coverlet.collector"
    },
    {
      "Id": "xunit",
      "Name": "xunit"
    },
    {
      "Id": "xunit.runner.visualstudio",
      "Name": "xunit.runner.visualstudio"
    }
  ],
  "References": [
    {
      "From": "VsTestIssue\\DotNetStandardLibrary\\DotNetStandardLibrary.csproj",
      "To": "Newtonsoft.Json"
    },
    {
      "From": "VsTestIssue\\DotNetStandardLibrary.Tests\\DotNetStandardLibrary.Tests.csproj",
      "To": "VsTestIssue\\DotNetStandardLibrary\\DotNetStandardLibrary.csproj"
    },
    {
      "From": "VsTestIssue\\DotNetStandardLibrary.Tests\\DotNetStandardLibrary.Tests.csproj",
      "To": "Microsoft.NET.Test.Sdk"
    },
    {
      "From": "VsTestIssue\\DotNetStandardLibrary.Tests\\DotNetStandardLibrary.Tests.csproj",
      "To": "MSTest.TestAdapter"
    },
    {
      "From": "VsTestIssue\\DotNetStandardLibrary.Tests\\DotNetStandardLibrary.Tests.csproj",
      "To": "MSTest.TestFramework"
    },
    {
      "From": "VsTestIssue\\DotNetStandardLibrary.Tests\\DotNetStandardLibrary.Tests.csproj",
      "To": "coverlet.collector"
    },
    {
      "From": "VsTestIssue\\DotNetStandardLibrary.XUnit.Tests\\DotNetStandardLibrary.XUnit.Tests.csproj",
      "To": "VsTestIssue\\DotNetStandardLibrary\\DotNetStandardLibrary.csproj"
    },
    {
      "From": "VsTestIssue\\DotNetStandardLibrary.XUnit.Tests\\DotNetStandardLibrary.XUnit.Tests.csproj",
      "To": "Microsoft.NET.Test.Sdk"
    },
    {
      "From": "VsTestIssue\\DotNetStandardLibrary.XUnit.Tests\\DotNetStandardLibrary.XUnit.Tests.csproj",
      "To": "xunit"
    },
    {
      "From": "VsTestIssue\\DotNetStandardLibrary.XUnit.Tests\\DotNetStandardLibrary.XUnit.Tests.csproj",
      "To": "xunit.runner.visualstudio"
    },
    {
      "From": "VsTestIssue\\DotNetStandardLibrary.XUnit.Tests\\DotNetStandardLibrary.XUnit.Tests.csproj",
      "To": "coverlet.collector"
    }
  ]
}
```

for reference the HTML output looks as below.

![image](https://user-images.githubusercontent.com/2970078/120996411-666ab800-c7da-11eb-8d35-b385f598927e.png)

**Comment by quasarea** [comment]
oh, I forgot about that switch :) and you right, it works fine on this projects, but not on others, let me try to cut down other sample, better tested this time ;)

**Comment by quasarea** [comment]
![image](https://user-images.githubusercontent.com/10334982/121048332-9b611580-c7ae-11eb-98e2-ecf8af7b595f.png)
[SplashScreen.zip](https://github.com/madushans/DependenSee/files/6610267/SplashScreen.zip)
```
DependenSee.exe -SourceFolder .\SplashScreen\ -OutputType ConsoleJson -IncludePackages
{
  "Projects": [
    {
      "Id": "SplashScreen\\SplashScreen.csproj",
      "Name": "SplashScreen"
    },
    {
      "Id": "SplashScreen.Tests\\SplashScreen.Tests.csproj",
      "Name": "SplashScreen.Tests"
    }
  ],
  "Packages": [],
  "References": []
}
```

**Comment by madushans** [comment]
OK I can repro this now.
Part of is likely due to it being .NET Framework projects which is not quite supported at the moment. I will further test it to confirm.

As this seem like a fairly simple case, this will likely result in a new release with a fix or support for .NET Framework projects.

Will keep this issue open until this is fixed.

**Comment by madushans** [comment]
This is now fixed with version [1.2.1](https://github.com/madushans/DependenSee/releases/tag/1.2.1) (PR #8)

Please run `dotnet tool update DependenSee --global` to update and confirm the issue is resolved.

Thanks for using DependenSee ♥

**Comment by quasarea** [comment]
Thanks, works fine
